### PR TITLE
fix: cache ajv schemas rather than ajv objects for schema validation

### PR DESCRIPTION
**Pull request by dbcfd**

### CODE REVIEW

**Comment by stbrody** [comment]
I'd like @ukstv to take a look at this also since he added the caching here originally and has looked into how Ajv does its internal caching in the past as well

**Review by stephhuynh18 — APPROVED** [review]

File: packages/stream-model-instance-handler/src/schema-utils.ts [hunk 1]
```diff
@@ -22,18 +19,17 @@ function buildAjv(): Ajv {
...
 
   public validateSchema(content: Record<string, any>, schema: SchemaObject, schemaId: string) {
-    let existingSchema = this.schemas.get(schemaId)
+    let existingSchema = this.ajv.getSchema(schemaId)
     if (!existingSchema) {
-      existingSchema = this.ajv.compile(schema)
-      this.schemas.set(schemaId, existingSchema)
+      this.ajv.addSchema(schema, schemaId)
+      //we've added the schema above, so ajv will have it
+      existingSchema = this.ajv.getSchema(schemaId)!
```
Comment (stbrody) [inline packages/stream-model-instance-handler/src/schema-utils.ts:32]: is the exclamation point at the end intentional? If so what is it doing?

Comment (dbcfd) [reply packages/stream-model-instance-handler/src/schema-utils.ts:32]: Resolving the possible null value as not null. We just inserted, so it should never be null

**Review by ukstv — APPROVED** [review]

**Review by stbrody — APPROVED** [review]
lgtm

**Pull request merged**